cpp/MPMCQueue.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <atomic>
+#include <new>
+
+#include "Vector.hpp"
+
+
+namespace hsd
+{
+    namespace priv
+    {
+        template<typename T>
+        struct Slot
+        {
+            void destroy()
+            {
+                reinterpret_cast<T*>(&storage)->~T();
```

Comment:
This is bad, you know you can write `storage.~T();`

Reply:
I was writing with C++11 in mind, I'll fix that.